aries_vcx/tests/test_integration.rs

```diff
@@ -1001,38 +1014,104 @@ mod tests {
+
         institution.activate().unwrap();
-        let public_invite_json = institution.create_public_invite();
-        let public_invite: Invitation = serde_json::from_str(&public_invite_json).unwrap();
+
+        let (schema_id, _schema_json, cred_def_id, _cred_def_json, cred_def, rev_reg_id) = _create_address_schema();
+
+        let institution_did = settings::get_config_value(settings::CONFIG_INSTITUTION_DID).unwrap();
+        let _requested_attrs = requested_attrs(&institution_did, &schema_id, &cred_def_id, None, None);
+        let requested_attrs_string = serde_json::to_string(&_requested_attrs).unwrap();
+
+        let mut request_sender = create_proof_request(&mut institution, &requested_attrs_string, "[]", "{}", None);
+
+        let service = FullService::try_from(&institution.agent).unwrap();
+        let oob_sender = OutOfBand::create()
+            .set_label("test-label")
+            .set_goal_code(GoalCode::P2PMessaging)
+            .set_goal("To exchange message")
+            .append_service(ServiceResolvable::FullService(service))
+            .append_handshake_protocol(HandshakeProtocol::ConnectionV1).unwrap()
+            .append_a2a_message(request_sender.to_a2a_message()).unwrap();
+        let oob_msg = oob_sender.to_a2a_message();
 
         consumer.activate().unwrap();
-        let mut consumer_to_institution = Connection::create_with_invite("institution", public_invite, true).unwrap();
-        consumer_to_institution.connect().unwrap();
-        consumer_to_institution.update_state().unwrap();
+        let oob_receiver = OutOfBand::create_from_a2a_msg(&oob_msg).unwrap();
+        let conn = oob_receiver.connection_exists(vec![]).unwrap();
+        assert!(!conn.is_some());
+        let mut conn_receiver = oob_receiver.build_connection(true).unwrap();
+        conn_receiver.connect().unwrap();
+        conn_receiver.update_state().unwrap();
+        assert_eq!(ConnectionState::Invitee(InviteeState::Requested), conn_receiver.get_state());
+        assert_eq!(oob_sender.id.0, oob_receiver.id.0);
 
         institution.activate().unwrap();
         thread::sleep(Duration::from_millis(500));
-        let mut conn_requests = institution.agent.download_connection_requests(None).unwrap();
-        assert_eq!(conn_requests.len(), 1);
-        let mut institution_to_consumer = Connection::create_with_connection_request(conn_requests.pop().unwrap(), &institution.agent).unwrap();
-        assert_eq!(ConnectionState::Inviter(InviterState::Requested), institution_to_consumer.get_state());
-        institution_to_consumer.update_state().unwrap();
-        assert_eq!(ConnectionState::Inviter(InviterState::Responded), institution_to_consumer.get_state());
+        let mut requests = institution.agent.download_connection_requests(None).unwrap();
+        assert_eq!(requests.len(), 1);
+        let mut conn_sender = Connection::create_with_connection_request(requests.pop().unwrap(), &institution.agent).unwrap();
```

Comment:
Seems to be decent fragment of code duplicated in following lines and in `create_connected_connections_via_public_invite` - I think if `create_connected_connections_via_public_invite` is perhaps split in 2 functions / generalized, we could get rid of the duplicity. 

Following lines seems to be duplicate:
```
 let mut requests = institution.agent.download_connection_requests(None).unwrap();
        assert_eq!(requests.len(), 1);
        let mut conn_sender = Connection::create_with_connection_request(requests.pop().unwrap(), &institution.agent).unwrap();
        assert_eq!(ConnectionState::Inviter(InviterState::Requested), conn_sender.get_state());
        conn_sender.update_state().unwrap();
        assert_eq!(ConnectionState::Inviter(InviterState::Responded), conn_sender.get_state());

        consumer.activate().unwrap();
        thread::sleep(Duration::from_millis(500));
        conn_receiver.update_state().unwrap();
        assert_eq!(ConnectionState::Invitee(InviteeState::Completed), conn_receiver.get_state());

        institution.activate().unwrap();
        thread::sleep(Duration::from_millis(500));
        conn_sender.update_state().unwrap();
        assert_eq!(ConnectionState::Inviter(InviterState::Completed), conn_sender.get_state());
        ```
        
Not a big deal, consider some extraction - if it turns out problematic for some reason, or not very useful, nevermind.

Reply:
Yeah, agreed - there quite a bit of duplication in the integration tests in general.